server/conn.go

```diff
@@ -259,6 +259,41 @@ func (cc *clientConn) writeInitialHandshake() error {
 	return cc.flush()
 }
 
+func (cc *clientConn) ReadFromStream() ([]byte, error) {
+	return cc.readPacket()
+}
+
+func (cc *clientConn) ReqData(filePath string) error {
+	return cc.writeReq(filePath)
+}
+
+func (cc *clientConn) LoadPreCheck() error {
+	// If the server handles the load data request, the client has to set the ClientLocalFiles capability.
+	if cc.capability&mysql.ClientLocalFiles == 0 {
+		return errNotAllowedCommand
+	}
+	return nil
+}
+
+func (cc *clientConn) TxnOp(ctx context.Context, loadDataInfo *executor.LoadDataInfo, err error) error {
+	var txn kv.Transaction
+	var err1 error
+	txn, err1 = loadDataInfo.Ctx.Txn(true)
+	if err1 == nil {
+		if txn != nil && txn.Valid() {
+			if err != nil {
+				if err1 := txn.Rollback(); err1 != nil {
+					logutil.Logger(ctx).Error("load data rollback failed", zap.Error(err1))
+				}
+				return err
+			}
+			return cc.ctx.CommitTxn(sessionctx.SetCommitCtx(ctx, loadDataInfo.Ctx))
+		}
+	}
+	// Should never reach here.
```

Comment:
Why???
If get tso fail, `Txn()` could return error...

Reply:
I'm not quit sure this logic, same with old clientConn.handleLoadData
@jackysp  PTAL do we need to panic here, or is there some historical reasons?